# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/util/ActiveElementManager.h
# Commit: e974d872becd
# Full Hash: e974d872becd4fae0214738cdf27173a994d0f59
# Author: Dan Robertson <drobertson@mozilla.com>
# Date: 2023-08-30 21:27:31
# Regressor Bug: 1816473
# File Overlap Count: 1
# Description:
#   Bug 1816473 - Do not delay dispatching synthesized mouse events. r=botond,edgar
#   
#   Do not delay dispatching synthesized mouse events for a single tap
#   gesture.
#   
# ==============================================================================

diff -r 1b272e14e402 -r e974d872becd gfx/layers/apz/util/ActiveElementManager.h
--- a/gfx/layers/apz/util/ActiveElementManager.h	Wed Aug 30 12:06:41 2023 +0000
+++ b/gfx/layers/apz/util/ActiveElementManager.h	Wed Aug 30 12:49:22 2023 +0000
@@ -21,6 +21,8 @@
 
 namespace layers {
 
+class DelayedClearElementActivation;
+
 /**
  * Manages setting and clearing the ':active' CSS pseudostate in the presence
  * of touch input.
@@ -61,6 +63,14 @@
    * delayed until after touch listeners have responded to the APZ.
    */
   void HandleTouchEnd();
+  /**
+   * Possibly clear active element sate in response to a single tap.
+   */
+  void ProcessSingleTap();
+  /**
+   * Cleanup on window destroy.
+   */
+  void Destroy();
 
  private:
   /**
@@ -82,6 +92,10 @@
    */
   RefPtr<CancelableRunnable> mSetActiveTask;
 
+  // Store the pending single tap event element activation clearing
+  // task.
+  RefPtr<DelayedClearElementActivation> mDelayedClearElementActivation;
+
   // Helpers
   void TriggerElementActivation();
   void SetActive(dom::Element* aTarget);